patches/server/0965-Make-banners-modifiable-with-both-BannerMeta-and-Blo.patch

```diff
@@ -0,0 +1,270 @@
+ 
+     @Override
+     public DyeColor getBaseColor() {
++        if (true) return this.bannerBlock.getBaseColor(); // Paper - Use underlying block state
+         return this.base;
+     }
+ 
+     @Override
+     public void setBaseColor(DyeColor color) {
++        // Paper start - Use underlying block state
++        if (true) {
++            this.bannerBlock.setBaseColor(color);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return;
++        }
++        // Paper end - Use underlying block state
+         this.base = color;
+     }
+ 
+     @Override
+     public List<Pattern> getPatterns() {
++        if (true) return this.bannerBlock.getPatterns(); // Paper - Use underlying block state
+         return new ArrayList<Pattern>(this.patterns);
+     }
+ 
+     @Override
+     public void setPatterns(List<Pattern> patterns) {
++        // Paper start - Use underlying block state
++        if (true) {
++            this.bannerBlock.setPatterns(patterns);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return;
++        }
++        // Paper end - Use underlying block state
+         this.patterns = new ArrayList<Pattern>(patterns);
+     }
+ 
+     @Override
+     public void addPattern(Pattern pattern) {
++        // Paper start - Use underlying block state
++        if (true) {
++            this.bannerBlock.addPattern(pattern);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return;
++        }
++        // Paper end - Use underlying block state
+         this.patterns.add(pattern);
+     }
+ 
+     @Override
+     public Pattern getPattern(int i) {
++        if (true) return this.bannerBlock.getPattern(i); // Paper - Use underlying block state
+         return this.patterns.get(i);
+     }
+ 
+     @Override
+     public Pattern removePattern(int i) {
++        // Paper start - Use underlying block state
++        if (true) {
++            Pattern removedPattern = this.bannerBlock.removePattern(i);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return removedPattern;
++        }
++        // Paper end - Use underlying block state
+         return this.patterns.remove(i);
+     }
+ 
+     @Override
+     public void setPattern(int i, Pattern pattern) {
++        // Paper start - Use underlying block state
++        if (true) {
++            this.bannerBlock.setPattern(i, pattern);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return;
++        }
++        // Paper end - Use underlying block state
+         this.patterns.set(i, pattern);
+     }
+ 
+     @Override
+     public int numberOfPatterns() {
++        if (true) return this.bannerBlock.numberOfPatterns(); // Paper - Use underlying block state
+         return this.patterns.size();
+     }
```

Comment:
Same thing here, just delegate directly to the get/setBlockState methods instead of using a copy of the blockstate stored on the banner meta.